### PR TITLE
[Twilio Adapter] Decouple Process Async

### PR DESCRIPTION
**Pull request by VictorGrycuk**
- Create a method to set `HttpResponse` using a `Context`

### CODE REVIEW

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs [hunk 1]
```diff
@@ -257,5 +258,35 @@ public void ValidateRequest_Should_Fail_With_NonMatching_Signature()
...
+            var twilioApi = new Mock<ITwilioClient>();
+            twilioApi.Setup(x => x.GetResourceIdentifier(It.IsAny<object>())).Returns(Task.FromResult(resourceIdentifier));
+
+            // Create a new Twilio Adapter with the mocked classes and get the responses
+            var twilioAdapter = new TwilioAdapter(options.Object, twilioApi.Object);
+            var activity = new Mock<Activity>().SetupAllProperties();
+
+            using (var context = new TurnContext(twilioAdapter, activity.Object))
+            {
+                context.TurnState.Add("httpStatus", "200");
```
Comment (GasparAcevedoZainSouthworks) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs]: Replace `200` with `HttpStatusCode.OK.ToString("D")`

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs [hunk 2]
```diff
@@ -257,5 +258,35 @@ public void ValidateRequest_Should_Fail_With_NonMatching_Signature()
...
+                var httpResponse = new Mock<HttpResponse>().SetupAllProperties();
+
+                TwilioHelper.SetHttpResponse(context, httpResponse.Object);
+
+                Assert.Equal(200, httpResponse.Object.StatusCode);
```
Comment (GasparAcevedoZainSouthworks) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs]: Replace `200` with `HttpStatusCode.OK.ToString("D")`

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs [hunk 3]
```diff
@@ -257,5 +258,35 @@ public void ValidateRequest_Should_Fail_With_NonMatching_Signature()
...
+            options.Object.AuthToken = "MockToken";
+            options.Object.TwilioNumber = "MockNumber";
+
+            // Setup mocked Twilio API client
+            const string resourceIdentifier = "Mocked Resource Identifier";
+            var twilioApi = new Mock<ITwilioClient>();
+            twilioApi.Setup(x => x.GetResourceIdentifier(It.IsAny<object>())).Returns(Task.FromResult(resourceIdentifier));
+
+            // Create a new Twilio Adapter with the mocked classes and get the responses
+            var twilioAdapter = new TwilioAdapter(options.Object, twilioApi.Object);
```
Comment (GasparAcevedoZainSouthworks) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs]: _Future refactor_: extract the creation of this _TwilioAdapter_ (including the _options_ and the _twilioApi_) to another method.

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs [hunk 4]
```diff
@@ -257,5 +258,35 @@ public void ValidateRequest_Should_Fail_With_NonMatching_Signature()
                 TwilioHelper.RequestToActivity(httpRequest.Object, string.Empty, string.Empty);
             });
         }
+
+        [Fact]
+        public void Test()
```
Comment (GasparAcevedoZainSouthworks) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs]: Use another name for the _test_

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs [hunk 5]
```diff
@@ -103,6 +103,14 @@ public static Activity RequestToActivity(HttpRequest httpRequest, string validat
             };
         }
 
+        public static HttpResponse SetHttpResponse(TurnContext context, HttpResponse httpResponse)
```
Comment (GasparAcevedoZainSouthworks) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs]: Add method's comments/description

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs [hunk 6]
```diff
@@ -103,6 +103,25 @@ public static Activity RequestToActivity(HttpRequest httpRequest, string validat
...
 
+        /// <summary>
+        /// Sets the StatusCode and ContentType of a HttpResponse using a TurnContext.
+        /// </summary>
+        /// <param name="context">The TurnContext that contains the HttpStatus.</param>
+        /// <param name="httpResponse">The HttpResponse to set the Status code and ContentType.</param>
+        /// <returns>HttpResponse with updated StatusCode and ContentType.</returns>
+        public static HttpResponse SetHttpResponse(TurnContext context, HttpResponse httpResponse)
+        {
+            if (!context.TurnState.Keys.Contains("httpStatus"))
```
Comment (GasparAcevedoZainSouthworks) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs]: We should check first if both _context_ and _httpResponse_ are not null

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs (hunk 3 above)
Comment (VictorGrycuk) [reply tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs]: I heartily agree. This change would greatly improve the readability of the code.

**Pull request closed without merging**